### DISCUSSION TITLE
Does git-release not trigger a "release" webhook event with "created" activity type?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [anton-yurchenko/git-release#100] [Q&A] [answered]
I have a CI workflow that listens to the [`created`](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=released#release) activity type for [`release`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) webhook events:

```yaml
name: "Sync"

on:
  release:
    types:
      - "created"
```

If I create a release manually via GitHub UI, this sync workflow runs as expected. However, this workflow doesn't work when the releases are made via `git-release`. Here's a link to my `release.yml` workflow:

https://github.com/PaulRBerg/prb-test/blob/d02ee246d8e371d02d6c4c2e72fbe8df9d94fe11/.github/workflows/release.yml

Is this normal? Does `git-release` not trigger a `created` activity type? Should I switch to using `published`?

**Update**: `published` didn't work either. It looks like releases generated by `github-actions` don't trigger a `release` webhook event at all?

**Comment by PaulRBerg** [comment, marked as answer]
I have figured out the answer on my own by arriving at this [StackOverflow Q&A](https://stackoverflow.com/q/73486033/3873510) and talking to ChatGPT4.

Turns out that GitHub Actions [prevents recursive workflows](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token) by design.

Here's the full response given by ChatGPT4:

````markdown
Yes, it is normal for releases made by `github-actions` to not trigger a `release` webhook event when the action is running in the same repository. This is because GitHub Actions [prevents recursive workflows](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token) by design.

To work around this limitation, you can use a personal access token (PAT) to authenticate when creating the release, which will allow the `release` webhook event to trigger other workflows in the same repository. Here's how to do it:

1. Create a personal access token (PAT) with the `repo` scope following [these instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).

2. Add the PAT to your repository secrets. Go to the "Settings" tab of your repository, click on "Secrets" in the left sidebar, and then click the "New repository secret" button. Give it a name like `PAT` and paste the token you created in step 1.

3. Update your `release.yml` workflow to use the PAT instead of the default `GITHUB_TOKEN`. You can do this by modifying the `env` section of the `git-release` step:

```yaml
steps:
  - name: Git Release
    uses: anton-yurchenko/git-release@v3.1.1
    env:
      GITHUB_TOKEN: ${{ secrets.PAT }}
    with:
      ...
```

Now, when the `git-release` action creates a release, it should trigger your `Sync` workflow as expected, since it will be using your personal access token instead of the default `GITHUB_TOKEN`.

````

**Reply by anton-yurchenko** [reply]
Lovely assistant 🤩
